sub/mutation_events.go

```diff
@@ -54,16 +54,19 @@ func MutationEventHandlerToStringHandler[T proto.Message](handler ProtoMutationE
 			return fmt.Errorf("error unmarshaling sns mutation event. why=%w", err)
 		}
 
+		//only unmarshal the before and after field if there are  not nil
 		before := newMessage()
-		err = protojson.Unmarshal([]byte(msg.Before), before)
-		if err != nil {
-			return fmt.Errorf("error unmarshaling 'Before' field from sns mutation event. why=%w", err)
+		if msg.Before != nil {
+			if err := protojson.Unmarshal(msg.Before, before); err != nil {
+				return fmt.Errorf("error unmarshaling 'Before' field from sns mutation event. why=%w", err)
+			}
 		}
 
 		after := newMessage()
```

Comment:
One issue with this implementation is that the `before` and `after` variables are still initialized, which means we end up passing **empty** values, not _nil_ values. It's not unlikely that the handler will do soemthing like this:
```
if before != nil {
   // get some values from before
}
```